internal/watcher/health/health_watcher_service.go

```diff
@@ -80,6 +80,16 @@ func (hw *HealthWatcherService) DeleteHealthWatcher(instances []*mpi.Instance) {
 	}
 }
 
+func (hw *HealthWatcherService) GetInstancesHealth() []*mpi.InstanceHealth {
```

Comment:
Probably need to add a `sync.Mutex` to put a lock around the cache since the health watcher server could be updating it at the same time you are reading from it.

Reply:
Great point, it's updated.